core/src/io/anuke/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -299,4 +317,27 @@ protected void drawLaser(Tile tile, Tile target){
         Draw.color();
     }
 
+    public static boolean insulated(Tile tile, Tile other){
+        return insulators(tile, other).size > 0;
+    }
+
+    public static Array<Tile> insulators(Tile tile, Tile other) {
+        Array<Tile> tiles = new Array<Tile>();
+        Array<Point2>points = PlaceUtils.normalizeDiagonal(tile.x, tile.y, other.x, other.y);
```

Comment:
This is a very... strange place to use this method. It gives unclean diagonals, which can be seen in your GIF.

How I would do this method:
```java
public static Array<Tile> insulators(int x, int y, int x2, int y2, Consumer<Tile> iterator){
    world.raycastEach(x, y, x2, y2, (wx, wy) -> {
      //do stuff to get the tile
     iterator.accept(world.ltile(wx, wy)); // if it fits the criteria
    });
}
```

Reply:
I had no clue on how to get `Array<Tile>` out of your example 😅